### PR TITLE
Add temporary option to allow suppression of recoverable LSP error toasts

### PR DESCRIPTION
**Pull request by dibarbet**
Resolves https://github.com/dotnet/vscode-csharp/issues/6977

Add temporary option to suppress LSP recoverable errors until we can resolve the architectural issues causing the failed requests.
Related to https://github.com/dotnet/vscode-csharp/issues/6973, https://github.com/microsoft/vscode-dotnettools/issues/722 and more

Implemented as suggested in https://github.com/microsoft/vscode-languageserver-node/issues/1449#issuecomment-2074436676

Additional notes
1.  VSCode's notification settings intentionally do not allow errors to be suppressed.
2.  Ideally we could look at VSCode's do not disturb settings or notification settings for the extension instead of our own option, but the API for that does not exist.

### CODE REVIEW

File: src/lsptoolshost/roslynLanguageClient.ts [hunk 1]
```diff
@@ -14,6 +21,8 @@ import { IDisposable } from '../disposable';
 export class RoslynLanguageClient extends LanguageClient {
     private readonly _disposables: CompositeDisposable;
 
+    //private readonly Map<
```
Comment (Cosifne) [inline src/lsptoolshost/roslynLanguageClient.ts:24]: 🤔Unused comment?

File: src/lsptoolshost/roslynLanguageClient.ts [hunk 2]
```diff
@@ -31,6 +40,27 @@ export class RoslynLanguageClient extends LanguageClient {
...
+    ) {
+        // Temporarily allow LSP error toasts to be suppressed if configured.
+        // There are a few architectural issues preventing us from solving some of the underlying problems,
+        // for example Razor cohosting to fix text mismatch issues and unification of serialization libraries
+        // to fix URI identification issues.  Once resolved, we should remove this option.
+        //
+        // See also https://github.com/microsoft/vscode-dotnettools/issues/722
+        // https://github.com/dotnet/vscode-csharp/issues/6973
+        // https://github.com/microsoft/vscode-languageserver-node/issues/1449
+        if (languageServerOptions.suppressLspErrorToasts) {
```
Comment (Cosifne) [inline src/lsptoolshost/roslynLanguageClient.ts:58]: 🤔how to know if the error is recoverable LSP error or not?

Comment (Cosifne) [reply src/lsptoolshost/roslynLanguageClient.ts:58]: Or is it just suppressing everything for a temp workaround?

File: src/lsptoolshost/roslynLanguageClient.ts (hunk 1 above)
Comment (dibarbet) [reply src/lsptoolshost/roslynLanguageClient.ts]: oops, fixed!

**Review by Cosifne — APPROVED** [review]

**Pull request merged**